rebasehelper/cli.py

```diff
@@ -123,168 +122,37 @@ class CLI(object):
-        )
-        parser.add_argument(
-            "--disable-inapplicable-patches",
-            default=False,
-            action="store_true",
-            dest='disable_inapplicable_patches',
-            help="disable inapplicable patches in rebased SPEC file"
-        )
-        parser.add_argument(
-            "--non-interactive",
-            default=False,
-            action="store_true",
-            dest='non_interactive',
-            help="do not interact with user"
-        )
-        parser.add_argument(
-            "--build-tasks",
-            dest="build_tasks",
-            metavar="OLD_TASK,NEW_TASK",
-            type=lambda s: s.split(','),
-            help="comma-separated remote build task ids"
-        )
-        parser.add_argument(
-            "--builds-nowait",
-            default=False,
-            action="store_true",
-            help="do not wait for remote builds to finish"
-        )
-        parser.add_argument(
-            "--builder-options",
-            default=None,
-            metavar="BUILDER_OPTIONS",
-            help="enable arbitrary local builder option(s), enclose %(metavar)s in quotes "
-                 "and note that = before it is mandatory"
-        )
-        parser.add_argument(
-            "--srpm-builder-options",
-            default=None,
-            metavar="SRPM_BUILDER_OPTIONS",
-            help="enable arbitrary local srpm builder option(s), enclose %(metavar)s in quotes "
-                 "and note that = before it is mandatory"
-        )
-        parser.add_argument(
-            "--results-dir",
-            help="directory where rebase-helper output will be stored"
-        )
-        parser.add_argument(
-            "--get-old-build-from-koji",
-            default=False,
-            action="store_true",
-            help="do not build old sources, download latest build from Koji instead"
-        )
-        parser.add_argument(
-            "sources",
-            metavar='SOURCES',
-            nargs='?',
-            default=None,
-            help="new upstream sources"
-        )
-        parser.add_argument(
-            "--changelog-entry",
-            default="- New upstream release %{version}",
-            help="text to use as changelog entry, can contain RPM macros, which will be expanded"
-        )
+        for node in OPTIONS:
+            if type(node) is list:
+                group = parser.add_mutually_exclusive_group()
+                for option in node:
+                    option_args = dict(option)
+                    for key in ("name", "default"):
+                        if key in option_args.keys():
+                            del option_args[key]
+                    group.add_argument(*option["name"], **option_args, action=CustomAction, default=SUPPRESS,
+                                       actual_default=option["default"] if "default" in option.keys() else None)
```

Comment:
This block (inside the `for` loop) is the same as below, if you want to keep it like this, you should define a function/method to avoid duplicate code. But it might be better to traverse options recursively.